public/pages/CreateSnapshotPolicy/components/CronSchedule/CronSchedule.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import _ from "lodash";
+import React, { ChangeEvent, useEffect, useState } from "react";
+import {
+  EuiCheckbox,
+  EuiComboBox,
+  EuiDatePicker,
+  EuiFieldNumber,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLink,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+} from "@elastic/eui";
+import CustomLabel from "../../../../components/CustomLabel";
+import { WEEK_DAYS, CRON_SCHEDULE_FREQUENCY_TYPE, TIMEZONES } from "./constants";
+import { buildCronExpression, parseCronExpression, startTime } from "./helper";
+import moment from "moment-timezone";
+import { CRON_EXPRESSION_DOCUMENTATION_URL } from "../../../../utils/constants";
+
+interface CronScheduleProps {
+  frequencyType: string;
+  onChangeFrequencyType: (e: ChangeEvent<HTMLSelectElement>) => void;
+  cronExpression: string;
+  onCronExpressionChange: (expression: string) => void;
+  showTimezone?: boolean;
+  timezone?: string;
+  onChangeTimezone?: (timezone: string) => void;
+  timezoneError?: string;
+}
+
+const CronSchedule = ({
+  frequencyType,
+  onChangeFrequencyType,
+  cronExpression,
+  onCronExpressionChange,
+  showTimezone,
+  timezone,
+  onChangeTimezone,
+  timezoneError,
+}: CronScheduleProps) => {
+  const { minute: initMin, hour: initHour, dayOfWeek: initWeek, dayOfMonth: initMonth } = parseCronExpression(cronExpression);
+
+  const [minute, setMinute] = useState(initMin);
+  const [hour, setHour] = useState(initHour);
+  const [dayOfWeek, setWeek] = useState(initWeek);
+  const [dayOfMonth, setMonth] = useState(initMonth);
+
+  useEffect(() => {
+    changeCron();
+  }, [minute, hour, dayOfWeek, dayOfMonth]);
+
+  const changeCron = (input?: any) => {
+    let cronParts = { hour, minute, dayOfWeek, dayOfMonth, frequencyType };
+    cronParts = { ...cronParts, ...input };
+    const expression = buildCronExpression(cronParts, cronExpression);
+    // console.log(`sm dev built expression ${expression}`);
```

Comment:
I see a lot of commented out lines of old code in this file

Reply:
Some of the comments like this are kept deliberately